python/cuspatial/cuspatial/geometry/geocolumn.py

```diff
@@ -30,23 +31,89 @@ def __init__(
         data: Tuple,
         meta: GeoMeta = None,
         shuffle_order: cudf.Index = None,
+        from_read_polygon_shapefile=False,
     ):
-        if (
-            not isinstance(data[0], cudf.Series)
-            or not isinstance(data[1], cudf.Series)
-            or not isinstance(data[2], cudf.Series)
-            or not isinstance(data[3], cudf.Series)
+        if from_read_polygon_shapefile:
+            polygons_col = data[0].astype("int32")
+            rings_col = data[1].astype("int32")
+            coordinates = (
+                data[2].stack().astype("float64").reset_index(drop=True)
+            )
+            coordinate_offsets = cudf.concat(
+                [
+                    cudf.Series(
+                        cp.arange(len(coordinates) // 2) * 2, dtype="int32"
+                    ),
+                    cudf.Series([len(coordinates) // 2 * 2], dtype="int32"),
```

Comment:
What is this arithmetic doing? Perhaps document it. (Also  below for the point offsets).

Reply:
Done